omicron/cli/process.py

```diff
@@ -783,12 +790,15 @@ def main(args=None):
 
     # -- find run segments
     # get segments from state vector
-    if (online and statechannel) or (statechannel and not stateflag) or (
-            statechannel and args.no_segdb):
+    if ((online and statechannel) or (statechannel and not stateflag) or (statechannel and args.no_segdb)) and \
+            dataduration <= 3600:
```

Comment:
It may be easier to read (and maintain) by splitting over many lines

```suggestion
    if dataduration <= 3600 and (
        (online and statechannel)
        or (statechannel and not stateflag)
        or (statechannel and args.no_segdb)
    ):
```

Reply:
But also, why 3600 seconds, why not 1800, or 7200?

Reply:
Does this effectively _forbid_ using a statevector when processing more than 1 hour of data?

Reply:
This has been a troublesome issue that I'm still searching for a solution.
The original problem was that the segment file was updated regardless of whether the online job succeeded. So when we had delays we didn't process anything. Now the problem is if we get too  far behind, a way too common occurrence we get stuck reading hours of raw frame files until evicted with no checkpointing.

I'm guessing at a robust limit so perhaps the "solution" is to make it another command line argument.